obs/packages.go

```diff
@@ -0,0 +1,194 @@
+	Name        string   `json:"name" xml:"name,attr"`
+	Title       string   `json:"title,omitempty" xml:"title,omitempty"`
+	Description string   `json:"description,omitempty" xml:"description,omitempty"`
+	Devel       Devel    `json:"devel" xml:"devel"`
+	Entries     []Entry  `json:"entry" xml:"entry"`
+}
+
+// Devel represents the development information
+type Devel struct {
+	Project string `json:"project" xml:"project,attr"`
+	Package string `json:"package" xml:"package,attr"`
+}
+
+// Entry is a struct representing an individual package entry.
+type Entry struct {
+	Name string `json:"name" xml:"name,attr"`
+}
+
+// CreateUpdatePackage creates a new OBS package or updates an existing OBS package of a project
+func (o *OBS) CreateUpdatePackage(ctx context.Context, projectName string, pkg *Package) error {
+	xmlData, err := xml.MarshalIndent(pkg, "", " ")
+	if err != nil {
+		return fmt.Errorf("creating obs package: marshalling package meta: %w", err)
+	}
+
+	urlPath, err := url.JoinPath(o.options.APIURL, "source", projectName, pkg.Name, "_meta")
+	if err != nil {
+		return fmt.Errorf("creating obs package: joining url: %w", err)
+	}
+
+	resp, err := o.client.InvokeOBSEndpoint(ctx, http.MethodPut, urlPath, bytes.NewBuffer(xmlData))
+	if err != nil {
+		return fmt.Errorf("creating obs package: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		var status Status
+		if err := xml.NewDecoder(resp.Body).Decode(&status); err != nil {
+			return &APIError{
+				HTTPStatusCode: resp.StatusCode,
+				OBSStatusCode:  "",
+				Message:        fmt.Sprintf("creating obs package: decoding error response: %v", err),
+			}
+		}
+
+		return &APIError{
+			HTTPStatusCode: resp.StatusCode,
+			OBSStatusCode:  status.Code,
+			Message:        status.Summary,
+		}
+	}
+
+	return nil
+}
+
+// ListPackages lists packages in a given OBS project
+func (o *OBS) ListPackages(ctx context.Context, projectName string) (*Package, error) {
+	urlPath, err := url.JoinPath(o.options.APIURL, "source", projectName)
+	if err != nil {
+		return nil, fmt.Errorf("listing obs packages: joining url: %w", err)
+	}
+
+	resp, err := o.client.InvokeOBSEndpoint(ctx, http.MethodGet, urlPath, nil)
+	if err != nil {
+		return nil, fmt.Errorf("listing obs packages: invoking obs endpoint: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		var status Status
+		if err := xml.NewDecoder(resp.Body).Decode(&status); err != nil {
+			return nil, &APIError{
+				HTTPStatusCode: resp.StatusCode,
+				OBSStatusCode:  "",
+				Message:        fmt.Sprintf("listing obs packages: decoding error response: %v", err),
+			}
+		}
+
+		return nil, &APIError{
+			HTTPStatusCode: resp.StatusCode,
+			OBSStatusCode:  status.Code,
+			Message:        status.Summary,
+		}
+	}
+
+	pkg := &Package{}
+	if err := xml.NewDecoder(resp.Body).Decode(&pkg); err != nil {
+		return nil, fmt.Errorf("listing obs packages: decoding packages: %w", err)
+	}
```

Comment:
This is a bit strange. This function is supposed to return a list of packages, but we're decoding the output into a single package. Can you please double check what is returned by the API?

Reply:
How about now?